tests/network/services.go

```diff
@@ -0,0 +1,297 @@
+		})
+
+		Context("with a service matching the vmi exposed", func() {
+			var isDualStack bool
+			var ipv6ServiceName string
+			var ipv6ServicePort int
+
+			BeforeEach(func() {
+				serviceName = "myservice"
+
+				service := buildServiceSpec(serviceName, servicePort, servicePort, selectorLabelKey, selectorLabelValue)
+				_, err := virtClient.CoreV1().Services(inboundVMI.Namespace).Create(service)
+				Expect(err).ToNot(HaveOccurred())
+
+				isDualStack = IsDualStackNetworkConfig()
+				if isDualStack {
+					ipv6ServiceName = serviceName + "v6"
+					ipv6ServicePort = servicePort + 6
+
+					service := buildIPv6ServiceSpec(ipv6ServiceName, servicePort, ipv6ServicePort, selectorLabelKey, selectorLabelValue)
+					_, err := virtClient.CoreV1().Services(inboundVMI.Namespace).Create(service)
+					Expect(err).ToNot(HaveOccurred())
+				}
+			})
+
+			AfterEach(func() {
+				deleteServiceResult := virtClient.CoreV1().Services(inboundVMI.Namespace).Delete(serviceName, &v13.DeleteOptions{})
+
+				if isDualStack {
+					deleteIPv6ServiceResult := virtClient.CoreV1().Services(inboundVMI.Namespace).Delete(ipv6ServiceName, &v13.DeleteOptions{})
+					Expect(deleteIPv6ServiceResult).To(Succeed())
+				}
+				Expect(deleteServiceResult).To(Succeed())
+			})
+
+			It("[test_id:XYZ] should be able to reach the vmi based on labels specified on the vmi", func() {
+				assertConnectionToExposedService(serviceName, inboundVMI.Namespace, servicePort)
+				if isDualStack {
+					assertConnectionToExposedService(ipv6ServiceName, inboundVMI.Namespace, ipv6ServicePort)
+				}
+			})
+
+			It("[test_id:1548]should fail to reach the vmi if an invalid servicename is used", func() {
+				assertConnectionToExposedServiceFails("wrongservice", inboundVMI.Namespace, servicePort)
+			})
+		})
+	})
+})
+
+func buildHeadlessServiceSpec(serviceName string, exposedPort int, portToExpose int, selectorKey string, selectorValue string) *v12.Service {
+	service := buildServiceSpec(serviceName, exposedPort, portToExpose, selectorKey, selectorValue)
+	service.Spec.ClusterIP = v12.ClusterIPNone
+	return service
+}
+
+func buildIPv6ServiceSpec(serviceName string, exposedPort int, portToExpose int, selectorKey string, selectorValue string) *v12.Service {
+	service := buildServiceSpec(serviceName, exposedPort, portToExpose, selectorKey, selectorValue)
+	ipv6Family := v12.IPv6Protocol
+	service.Spec.IPFamily = &ipv6Family
+
+	return service
+}
+
+func buildServiceSpec(serviceName string, exposedPort int, portToExpose int, selectorKey string, selectorValue string) *v12.Service {
+	return &v12.Service{
+		ObjectMeta: v13.ObjectMeta{
+			Name: serviceName,
+		},
+		Spec: v12.ServiceSpec{
+			Selector: map[string]string{
+				selectorKey: selectorValue,
+			},
+			Ports: []v12.ServicePort{
+				{Protocol: v12.ProtocolTCP, Port: int32(portToExpose), TargetPort: intstr.FromInt(exposedPort)},
+			},
+		},
+	}
+}
+
+func IsDualStackNetworkConfig() bool {
```

Comment:
It is quite probable this ends up being re-used elsewhere.

Where do you advice to put this @EdDev ? (I can just leave it here, and whoever needs it, figures that out).

Reply:
Hm, I didn't think this through; I think I can just expose a flag named `IsDualStackNetworkConfig`, that will be accessible throughout the tests.

The function is not needed, afaiu.